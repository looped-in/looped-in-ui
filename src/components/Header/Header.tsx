import Image from "next/image"
import React from "react"
import Button from "../Button"
import Link from "next/link"
import { HEADER_LINKS, PAGE_ROUTE } from "@/utils/constant"
import { usePathname } from "next/navigation"
import { cn } from "@/utils"

const Header = () => {
  const pathName = usePathname()

  return (
    <div
      className={cn("p-8", pathName === PAGE_ROUTE.HOMEPAGE ? "bg-white" : "")}
    >
      <div className="flex flex-row place-content-between">
        <Link href="/">
          <Image src="/logo.svg" alt="logo" width={150} height={150} />
        </Link>

        {/* Menu */}
        <div className="flex flex-row space-x-6 align-middle content-around text-sm font-semibold">
          {HEADER_LINKS.map(({ label, href }) => (
            <Link
              href={href}
              key={href}
              className={cn(
                "content-around hover:text-[#ff7338] text-black",
                pathName === href && "text-[#ff7338] "
              )}
            >
              {label}
            </Link>
          ))}

          <div className="content-around align-middle hover:text-[#ff7338]">
            <Link href={PAGE_ROUTE.SIGN_IN}>
              <Button label="Sign in" />
            </Link>
          </div>
        </div>
      </div>
    </div>
  )
}

export default Header
