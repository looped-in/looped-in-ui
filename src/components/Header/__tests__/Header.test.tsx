import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Header from "../Header";

describe("Header", () => {
	it("SHOULD match snapshot", () => {
		const { asFragment } = render(<Header />);
		expect(asFragment()).toMatchSnapshot();
	});

	it("renders the logo", () => {
		render(<Header />);
		const logo = screen.getByAltText("logo");
		expect(logo).toBeInTheDocument();
	});

	it("renders all navigation links", () => {
		render(<Header />);
		const links = [
			{ href: "/", text: "About" },
			{ href: "/how-it-work", text: "How it works" },
			{ href: "/our-team", text: "Our Team" },
			{ href: "/find-builder", text: "Find a builder" },
			{ href: "/get-in-touch", text: "Get in touch" },
			{ href: "/sign-in", text: "Sign in" },
		];

		for (const link of links) {
			const anchor = screen.getByText(link.text);
			expect(anchor).toBeInTheDocument();
			expect(anchor.closest("a")).toHaveAttribute("href", link.href);
		}
	});

	it("renders the Sign in button", () => {
		render(<Header />);
		const signInButton = screen.getByText("Sign in");
		expect(signInButton).toBeInTheDocument();
	});
});
