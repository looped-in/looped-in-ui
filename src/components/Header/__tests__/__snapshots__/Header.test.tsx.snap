// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header SHOULD match snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-16"
  >
    <div
      class="flex flex-row place-content-between"
    >
      <img
        alt="logo"
        data-nimg="1"
        decoding="async"
        height="150"
        loading="lazy"
        src="/logo.svg"
        style="color: transparent;"
        width="150"
      />
      <div
        class="flex flex-row space-x-6 align-middle content-around text-sm font-semibold"
      >
        <a
          class="content-around hover:text-[#ff7338]"
          href="/"
        >
          About
        </a>
        <a
          class="content-around hover:text-[#ff7338]"
          href="/how-it-work"
        >
          How it works
        </a>
        <a
          class="content-around hover:text-[#ff7338]"
          href="/our-team"
        >
          Our Team
        </a>
        <a
          class="content-around hover:text-[#ff7338]"
          href="/find-builder"
        >
          Find a builder
        </a>
        <a
          class="content-around hover:text-[#ff7338]"
          href="/get-in-touch"
        >
          Get in touch
        </a>
        <div
          class="content-around align-middle hover:text-[#ff7338]"
        >
          <a
            href="/sign-in"
          >
            <button
              class="h-10 w-32 rounded-tl-lg rounded-tr-lg rounded-br-lg bg-gradient-to-b from-[#ffb945] to-[#ff7338] text-white font-bold"
            >
              Sign in
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
